pyaedt/edb_grpc/core/edb_data/components_data.py

```diff
@@ -0,0 +1,962 @@
+        if num == 0:  # pragma: no cover
+            return None
+        elif num == 1:
+            return list(self.components.values())[0].type
+        else:
+            return "mixed"  # pragma: no cover
+
+    @type.setter
+    def type(self, value):
+        for comp in list(self.components.values()):
+            comp.type = value
+
+    @property
+    def components(self):
+        """Get the list of components belonging to this component definition.
+
+        Returns
+        -------
+        dict of :class:`pyaedt.edb_grpc.core.edb_data.components_data.EDBComponent`
+        """
+        comp_list = [
+            EDBComponent(self._pcomponents, l)
+            for l in self._pcomponents._edb.Cell.Hierarchy.Component.FindByComponentDef(
+                self._pcomponents._pedb.active_layout, self.part_name
+            )
+        ]
+        return {comp.refdes: comp for comp in comp_list}
+
+    @pyaedt_function_handler
+    def assign_rlc_model(self, res, ind, cap, is_parallel=False):
+        """Assign RLC to all components under this part name.
+
+        Parameters
+        ----------
+        res : int, float
+            Resistance.
+        ind : int, float
+            Inductance.
+        cap : int, float
+            Capacitance.
+        is_parallel : bool, optional
+            Whether it is parallel or series RLC component.
+        """
+        for comp in list(self.components.values()):
+            res, ind, cap = res, ind, cap
+            comp.assign_rlc_model(res, ind, cap, is_parallel)
+        return True
+
+    @pyaedt_function_handler
+    def assign_s_param_model(self, file_path, model_name=None, reference_net=None):
+        """Assign S-parameter to all components under this part name.
+
+        Parameters
+        ----------
+        file_path : str
+            File path of the S-parameter model.
+        name : str, optional
+            Name of the S-parameter model.
+        Returns
+        -------
+
+        """
+        for comp in list(self.components.values()):
+            comp.assign_s_param_model(file_path, model_name, reference_net)
+        return True
+
+    @pyaedt_function_handler
+    def assign_spice_model(self, file_path, model_name=None):
+        """Assign Spice model to all components under this part name.
+
+        Parameters
+        ----------
+        file_path : str
+            File path of the Spice model.
+        name : str, optional
+            Name of the Spice model.
+        Returns
+        -------
+
+        """
+        for comp in list(self.components.values()):
+            comp.assign_spice_model(file_path, model_name)
+        return True
+
+
+class EDBComponent(object):
+    """Manages EDB functionalities for components.
+
+    Parameters
+    ----------
+    parent : :class:`pyaedt.edb_grpc.core.components.Components`
+        Inherited AEDT object.
+    component : object
+        Edb Component Object
```

Comment:
```suggestion
        EDB component object.
```
The data type should be a link to the object.